### PR TITLE
Upgrade yarn

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
None

#### What's this PR do?
Upgrades yarn to latest stable version

#### How should this be manually tested?
Build should pass

### CODE REVIEW

**Comment by aliceriot** [comment]
Unfortunately there are four places where the yarn version needs to be declared:

- in `package.json` in the `engines` object: this is what Heroku uses to decide what version to install
- in `travis/Dockerfile-travis-watch-build`: this sets the version for running the tests
- in `Dockerfile-node`: for local development
- in the `README.md`: just for documentation

it would be great / awesome to figure out how to consolidate those, maybe we could use something like https://github.com/marcioj/check-pkg-engines with a little node.js script...idk!

**Comment by aliceriot** [comment]
I just spent a minute fiddling and came up with this:

```js
const fs = require('fs')
const { spawn } = require('child_process')

const { engines: { yarn: yarnVersion }} = JSON.parse(fs.readFileSync("package.json"))

let install = spawn('npm', [
  'install',
  '-g',
  `yarn@${yarnVersion}`
])

install.stdout.on('data', data => console.log(`${data}`))

install.stderr.on('data', err => console.log(`${err}`))

install.on('close', code => {
  if ( code === 0 ) {
    console.log('yarn installed successfully!')
  } else {
    console.err(`error: code ${code}`)
    console.err('\ndid you run as root?')
  }
})
```

**Comment by noisecapella** [comment]
How's this?

File: README.md [hunk 1]
```diff
@@ -167,10 +167,10 @@ look for the webpack server at that host instead of the host where Docker is run
 You'll need to install the [yarn](https://yarnpkg.com/en/docs/cli/)
 package manager. You can do:
 
-    npm install -g yarn@0.19.1
+    node ./scripts/install_yarn.js
```
Comment (aliceriot) [inline README.md]: I think most users will need to be root for this to work (`npm install -g`) but I am not user if nodejs would be installed for the user on MacOS.

File: travis/Dockerfile-travis-watch-build [hunk 2]
```diff
@@ -21,5 +19,8 @@ RUN chown -R mitodl:mitodl /src
 USER mitodl
 
 COPY package.json yarn.lock ./webpack_if_prod.sh /src/
+COPY scripts /src/scripts
+
+RUN node /src/install_yarn.js
```
Comment (aliceriot) [inline travis/Dockerfile-travis-watch-build]: this needs to be run as root, so can you copy the `USER mitodl` line below this?

the path to the script is going to be `/src/scripts/install_yarn.js`

File: README.md (hunk 1 above)
Comment (noisecapella) [inline README.md]: Sure

File: travis/Dockerfile-travis-watch-build (hunk 2 above)
Comment (noisecapella) [inline travis/Dockerfile-travis-watch-build]: Sure

**Pull request merged**